### ISSUE TITLE
Windows 10 Support

### ISSUE THREAD

**Issue by spawn-guy** [bitwarden/mobile#9]
Well, this whole thing is written in C# and Xamarin...

Why is there no W10/M version of this app? an UWP app maybe?

**Comment by kspearrin** [comment]
There are plans to add support for a UWP add. Shouldn't bee too much work to add.

**Comment by amoraitis** [comment]
Please assign this to me! I want to begin building Xamarin.UWP @kspearrin

**Comment by kspearrin** [comment]
@amoraitis Feel free to add a UWP project to the solution. This is Xamarin.Forms so the majority of the work should already be done.

**Comment by amoraitis** [comment]
@kspearrin Ok, i added a UWP project in src/! I want guidelines. I must add it in test first and commit changes there?

**Comment by kspearrin** [comment]
If you need some help with getting started I am always available for chat in our gitter channel: https://gitter.im/bitwarden/Lobby

**Comment by TotallyInformation** [comment]
+1 for this, especially as your website claims that it is for "all" devices. At the moment it clearly isn't.

**Comment by amoraitis** [comment]
Ok, to any dev: if you want to finish UWP project contribute to this forked repo:https://github.com/amoraitis/mobile and i will reopen the pull request here, once the project finish!

**Comment by Soneliem** [comment]
Any advancements on the Window 10 app? @amoraitis your github link does not work BTW.

**Comment by kspearrin** [comment]
@Soneliem No, unfortunately we have not made any progress here yet.

**Comment by Soneliem** [comment]
@kspearrin Sad to hear that. A windows app would be really useful.

**Comment by alistair23** [comment]
@amoraitis fork is down. How far did this end up getting?

**Comment by kspearrin** [comment]
@alistair23 As far as I know, nowhere.

**Comment by hichamboushaba** [comment]
Hello, I forked the repo, and added the necessary services to get the app running in Windows 10, we can continue working on the app on my fork until it's ready to make a pull request: https://github.com/hichamboushaba/bitwarden-mobile

Many services are still unimplemented, and the application needs a little theming, feel free to contribute, and me too I will try to implement the missing parts on my free time.

@kspearrin, I have some remarks/questions concerning the project:
1- you don't plan to migrate from PCL to Netstandard? this will give a boarded API which will help in more code sharing, I can help in this if you are interested
2- I had to add extensions to all image names in the PCL project, without them Forms doesn't resolve them for UWP. you are OK with this? AFAIK it doesn't impact other platforms (https://developer.xamarin.com/guides/xamarin-forms/user-interface/images/#Local_Images)
3- when do you plan to update forms nuget to v2.4?, it has a lot of fixes for UWP

Below some screenshots for the application in Windows:
![capture3](https://user-images.githubusercontent.com/1657201/30968789-57fa5338-a458-11e7-9bd3-2f9a3460ccc2.png)
![capture1](https://user-images.githubusercontent.com/1657201/30968790-5800ffda-a458-11e7-8fac-2933fc726a6f.PNG)
![capture2](https://user-images.githubusercontent.com/1657201/30968791-5802c504-a458-11e7-83fa-30c66c02831a.PNG)

**Comment by kspearrin** [comment]
@hichamboushaba That looks great so far! I glanced at your commits in your repo and it looks like you're definitely on the right track to get this working fully.

Your questions:

1. I'd like to and have attempted this a few times already but ran into several bugs, particularly around resource files for languages not working. Last time I tried was in VS 15.3 about 1-2 months ago. Maybe the tooling is fixed now with 15.4 preview (which I am using now).
2. I am fine with that as long as it doesn't affect other platforms. I can't think of any issues at the moment.
3. Looks like 2.4 went live yesterday. I'm fine with updating to it. Not sure what breaking changes it may have though.

**Comment by hichamboushaba** [comment]
@kspearrin, thanks, I will try to add the missing parts before making pull request

About the first point, I'm using Netstandard (csproj) for a personal project, and yes, the issue with resx files is fixed now.

**Comment by kspearrin** [comment]
@hichamboushaba Let's try to do this in phases. I'd like to keep the PRs as small as possible. So if you can avoid net standard and 2.4 for now, we can do that in another PR.

**Comment by hichamboushaba** [comment]
@kspearrin, yes of course :)
If you prefer I can even make a PR for what's done so far to keep it small.

**Comment by kspearrin** [comment]
If what you have compiles and runs that is good enough for a PR.

**Comment by alistair23** [comment]
I tested this over the weekend and it works on my Windows 10 Phone and Windows 10 computer. Are you planning on submitting the app to the Microsoft store soon?

**Comment by kspearrin** [comment]
It needs a lot of UI work before we can submit it.

**Comment by alistair23** [comment]
Ok, if you have specific areas in mind I can try and help.

**Comment by kspearrin** [comment]
Feel free to message me in our developer chatroom and I can go over a few things with you: https://gitter.im/bitwarden/Lobby

**Comment by alistair23** [comment]
Ok, so here are some main things that I see that still need to be done:

- [ ] Remove the fingerprint option (eventually this can probably be added back with Windows hello integration)
- [ ] Fix the camera not coming up when doing a QR scan
- [ ] Fix the vertical height being wrong with login and register pages. I think this needs a custom renderer, but I couldn't get it working even with that.
- [ ] Fix the Group vault not being displayed, all the others are it's just this one
- [ ] Fix the loading bar to go across the entire screen
- [ ] Fix the in app notifications (like copying a password) to look better

**Comment by alistair23** [comment]
I have forked the code base and have a bitwarden client available on the Windows store, if anyone is interested: https://www.microsoft.com/en-us/store/p/bytewarden/9n248v7dk05k
